### ISSUE TITLE
Support cleanup of deleted resources during live reload

### ISSUE THREAD

**Issue by Sanne** [quarkusio/quarkus#1772]
Taking for example the `import.sql` script of Hibernate ORM:

we produce a `new HotDeploymentConfigFileBuildItem(importFile)` for this file, so to allow people editing the script and applying those changes right away in the scope of "quarkus dev".

After #1736 it's now possible to start "dev mode", and then create the file later: it will be picked up.

The opposite isn't true though: if it's deleted after "dev mode" was already started, we don't clean such watched resources.

I haven't tested this with `.class` files but I suppose it applies as well.

**Comment by geoand** [comment]
I don't think the link to the PR is correct :)

**Comment by gsmet** [comment]
Class of interest if someone wants to take it: `RuntimeUpdatesProcessor`.

**Comment by Sanne** [comment]
thanks @geoand ! You're right, I meant #1736. Edited.

**Comment by geoand** [comment]
You are welcome @Sanne :)

**Comment by machi1990** [comment]
> Class of interest if someone wants to take it: `RuntimeUpdatesProcessor`.

@gsmet I'd like to give this one a shot. If I am understanding is correct, it involves cleaning the config timestamp registry and potentially removing the file from `target`?

**Comment by gsmet** [comment]
Yes, exactly, if a watched config file is not present anymore, we need to delete it.

As for classes, we have the same issue but it's not as simple as you don't have a 1-1 mapping between a source file and a class file (think of inner classes for instance).

I would deal with the config file in this issue and maybe open a new issue for the class files.

**Comment by machi1990** [comment]
@gsmet it is okay. Thanks for getting back. I'll look on this later tonight and propose a patch.

**Comment by gsmet** [comment]
I think this one is fully implemented now thanks to @machi1990 .
